diff --git a/qtconsole/console_widget.py b/qtconsole/console_widget.py
index 51e32ec..985ff6c 100644
--- a/qtconsole/console_widget.py
+++ b/qtconsole/console_widget.py
@@ -992,6 +992,7 @@ class ConsoleWidget(MetaQObjectHasTraits('NewBase', (LoggingConfigurable, superQ
 
         # Perform the insertion.
         result = insert(cursor, input, *args, **kwargs)
+        self._control.moveCursor(QtGui.QTextCursor.End)
         return result
 
     def _append_block(self, block_format=None, before_prompt=False):
